### PR TITLE
Add docs to half of the functions in `otel_span`

### PR DESCRIPTION
**Pull request by whatyouhide**
More docs. 🙃

### CODE REVIEW

File: apps/opentelemetry_api/src/otel_span.erl [hunk 1]
```diff
@@ -68,11 +70,15 @@ validate_start_opts(Opts) when is_map(Opts) ->
       is_recording => IsRecording
      }.
 
+%% @doc Returns whether the span is recording.
+%% @end
```
Comment (tsloughter) [inline apps/opentelemetry_api/src/otel_span.erl:74]:
```suggestion
```

File: apps/opentelemetry_api/src/otel_span.erl [hunk 2]
```diff
@@ -68,11 +70,15 @@ validate_start_opts(Opts) when is_map(Opts) ->
...
 -spec is_recording(SpanCtx) -> boolean() when
       SpanCtx :: opentelemetry:span_ctx().
 is_recording(SpanCtx) ->
     ?is_recording(SpanCtx).
 
+%% @doc Returns whether the span context is valid.
+%% @end
```
Comment (tsloughter) [inline apps/opentelemetry_api/src/otel_span.erl:81]:
```suggestion
```

**Comment by tsloughter** [comment]
Looks good but need consistency in whether `@end` is used or not. Then I can merge it.

**Comment by whatyouhide** [comment]
Done @tsloughter, thanks!

**Comment by tsloughter** [comment]
Still some functions (`add_event`, `end_span`) are including an explicit `@end` while others aren't.

**Comment by whatyouhide** [comment]
@tsloughter ah yes my bad, done for realsies now 🙃

**Review by tsloughter — APPROVED** [review]

**Pull request merged**